core/src/test/java/org/infinispan/stream/BaseSetupStreamIteratorTest.java

```diff
@@ -90,4 +100,52 @@ public String convert(Object key, String value, Metadata metadata) {
          }
       }
    }
+
+   private static class TestDefaultConsistentHashFactory
+         extends BaseControlledConsistentHashFactory<DefaultConsistentHash> {
+      TestDefaultConsistentHashFactory() {
+         super(new DefaultTrait(), 3);
+      }
+
+      @Override
+      protected List<Address> createOwnersCollection(List<Address> members, int numberOfOwners, int segmentIndex) {
+         if (members.size() < 3)
```

Comment:
you probably want to remove this condition otherwise members[0] will be always the primary owner of all the segments.